query-parser/src/parse.rs

```diff
@@ -213,6 +223,53 @@ def_value_parser_fn!(Where, or_join_clause, WhereClause, input, {
     }).parse_stream(input)
 });
 
+def_value_parser_fn!(Where, not_pattern_clause, WhereNotClause, input, {
+    Where::clause().map(|clause| WhereNotClause::Clause(clause)).parse_stream(input)
+});
+
+def_value_parser_fn!(Where, where_not_clause, WhereNotClause, input, {
+    choice([Where::not_pattern_clause()]).parse_stream(input)
+});
+
+def_value_parser_fn!(Where, not_clause, WhereClause, input, {
+    satisfy_map(|x: edn::Value| {
+        seq(x).and_then(|items| {
```

Comment:
I thought it landed and see now that it hasn't!  It's waiting on your final review: https://github.com/mozilla/mentat/pull/393#issuecomment-291260771.